### DISCUSSION TITLE
[RFC] Dependencies between cli packages

### DISCUSSION THREAD

**Discussion by beerose** [blitz-js/blitz#3682] [RFCs]
When I attempted to work on one of the codemod issues (https://github.com/blitz-js/blitz/issues/3601), I started looking into dependencies between packages. Especially between cli-related packages. In my case, I needed some utilities like fetch-latest-version, npm-fetch, etc., and they are currently in the `@blitzjs/generator` package. Though, in my opinion, they are quite general-purpose and not particularly related to the generator package. Ideally, in my opinion, I would see them as utilities in the `blitz` package for a few reasons:

1. This is our “core” package, where we have a bunch of utilities used by other packages
2. A while back, we decided that blitz will be this “core” package because other packages will have it as a dependency and use the “common” code defined in `blitz`.
3. I don’t think that some of the utilities in `@blitzjs/generator` are strictly related to the generator.

However, if we did that, there would be a circular dependency because `blitz` has `@blitzjs/generator` as a dependency. I get why it’s done this way, but I’d still like to see the utilities elsewhere. Especially since I wanted to use one of them in the `@blitzjs/codemod`, and I don’t like the idea of this kind of dependency between `@blitzjs/codemod` and `@blitzjs/generator` (as, in my opinion, they are separate packages that could (should?) exist separately. I can also imagine that there can be other utilities that we’ll want to keep in the core (`blitz`) package AND re-use in other packages like `@blitzjs/generator` (which currently is not possible).

See attached pictures for possible solutions.

<img width="572" alt="CleanShot 2022-08-05 at 12 44 19@2x" src="https://user-images.githubusercontent.com/9019397/183101716-cffe564a-f679-41cc-a78a-37fa66390cb0.png">

<img width="508" alt="CleanShot 2022-08-05 at 12 44 29@2x" src="https://user-images.githubusercontent.com/9019397/183101735-cfdf68d3-4826-421b-9643-39b6af33a1db.png">

<img width="507" alt="CleanShot 2022-08-05 at 12 44 36@2x" src="https://user-images.githubusercontent.com/9019397/183101731-493c504f-b842-4cd9-b46b-787ec44fd4ee.png">

<img width="607" alt="CleanShot 2022-08-05 at 12 44 42@2x" src="https://user-images.githubusercontent.com/9019397/183101728-75fe3765-f4cb-4ff1-b59e-2adc6b1b6291.png">

**Comment by flybayer** [comment]
Very nice, thank you!

I think I'd vote for solution 2, especially since `blitz` has the CLI and the generator stuff is only CLI, so it makes sense for all the CLI stuff to be in one package.

Second preference would be solution 3 but with the new package named `utils` or something.
